app/controllers/admin/deals_controller.rb

```diff
@@ -0,0 +1,55 @@
+module Admin
+  class DealsController < AdminController
+    before_action :set_deal, only: [:show, :edit, :update, :destroy]
+
+    def index
+      @deals = Deal.with_attached_images.all
```

Comment:
lets use https://github.com/kaminari/kaminari gem to show a paginated list to admin